zebra/zebra_vxlan.c

```diff
@@ -1990,12 +1990,23 @@ static int zvni_local_neigh_update(zebra_vni_t *zvni,
 	n = zvni_neigh_lookup(zvni, ip);
 	if (n) {
 		if (CHECK_FLAG(n->flags, ZEBRA_NEIGH_LOCAL)) {
+			bool check_rbit = false;
```

Comment:
This is good and works but I think this can done better.  If we modify as below, 

1) remove the return 0 condition as shown below.
2) Cover the remaining part of this if loop in else block, 

then I think we can avoid to have the boolean variable (check_rbit) and also all the related codes including goto statement. 

                       if (memcmp(n->emac.octet, macaddr->octet,
				   ETH_ALEN) == 0) {
				/* Update any params and return - client doesn't	
				 * care about a purely local change.		 */
				n->ifindex = ifp->ifindex;

                                  /* return 0; */
                           } else {
                              /* If the MAC has changed,
			       *   need to issue a delete first
			       * as this means a different MACIP route.
			       * Also, need to do some unlinking/relinking.
			       */
			        zvni_neigh_send_del_to_client(zvni->vni, &n->ip,
						      &n->emac, 0);
			         old_zmac = zvni_mac_lookup(zvni, &n->emac);
			          if (old_zmac) {
				      listnode_delete(old_zmac->neigh_list, n);
				      zvni_deref_ip2mac(zvni, old_zmac, 0);
                                    }
                           }

Reply:
Hi @srimohans, Thank you for reviewing PR.
 
Objective is to include neigh update with router flag change ( from set to unset vice-versa) but mac has not changed. Current condition (if statement) check if neigh's mac has not changed then simply "return".  I am trying to accommodate there router bit change from existing entry then don't "return" and still inform bgp with router flag, without sending delete request to BGPd. 

What you have propose would work for mac address same and still informs the bgpd which is redundant. So the boolean helps to update if only Router bit changed. 

I tried combining mac comparison and router flag check but both former is valid to "return" but latter requires to update bgpd. 

Hope it clarifies the intension.

Reply:
Chirag,

Understood the use of boolean variable. 

However, you can get rid of "goto" statement though with the above changes.